crates/cext/src/circuit_library/iqp.rs

```diff
@@ -0,0 +1,109 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use ndarray::ArrayView2;
+use qiskit_circuit::{circuit_data::CircuitData, operations::Param};
+use qiskit_circuit_library::iqp::{iqp, py_random_iqp};
+
+/// Internal helper: build `CircuitData` from a *validated* view.
+///
+/// Assumes:
+///   - `view` is square and symmetric.
+///   - Any error from `from_standard_gates` is considered unreachable for valid
+///     inputs, so we use `unwrap()` like other cext code.
+fn iqp_from_view(view : ArrayView2<'_, i64>) -> CircuitData {
```

Comment:
This looks like a legacy function from an earlier design -- why not just move this into the main function, since it's just 2 lines (or effectively one)?